src/storage/types.rs

```diff
@@ -183,12 +181,12 @@ impl Key {
         }
         if user_key_len >= number::U64_SIZE {
             // We compare last 8 bytes as u64 first, then compare the rest.
-            unsafe {
-                let ks_ptr = ts_encoded_key[user_key_len - 8..].as_ptr() as *const u64;
-                let rs_ptr = user_key[user_key_len - 8..].as_ptr() as *const u64;
-                if *ks_ptr != *rs_ptr {
-                    return false;
-                }
+            // TODO: Can we just use == to check the left part and right part? memcpy might be
```

Comment:
you mean `&ts_encoded_key[user_key_len - 8..ts_encoded_key.len() - 8]  == &user_key[user_key_len - 8..]` cause memcpy?

Reply:
Oh, sorry, it's a typo. It should be `memcmp`.

Reply:
Please fix it